lib/plugins/expressionParser.js

```diff
@@ -163,6 +165,19 @@ function expressionApplier(context, typeInformation) {
 
         newAttribute.value = applyExpression(attribute.expression, context, typeInformation);
 
+        if (attribute.type === 'Integer') {
+            newAttribute.value = Number.parseInt(newAttribute.value);
+        }
+        else if (attribute.type === 'Float') {
+            newAttribute.value = Number.parseFloat(newAttribute.value);
+        }
+        else if (attribute.type === 'Boolean') {
+            newAttribute.value = (newAttribute.value === true);
+        }
+        else if (attribute.type === 'Null') {
+            newAttribute.value = null;
+        }
```

Comment:
expressionLanguage.md should include information about how the mapping works in NGSIv2, i.e. how the different types (some of them news, e.g. Boolean, etc.) work. Maybe the best way by the time beinb would be a separate section (although in the future, if NGSIv1 is removed from IOTA library the entire .md should be rewritten).

Reply:
Fixed in d6581fc4915c900b66c0144df3f7d19bd11253b2

Reply:
I'm looking in detail de documentation and I read (with regards to NGSIv1):

> Expressions can have two return types: String or Number. This return type must be configured for each attribute that is going to be converted. Default value type is String.
>
> Whenever a expression is executed without error, its result will be cast to the configured type. If the conversion fails (e.g.: if the expression is null or a String and is casted to Number), the measurement update will fail, and an error will be reported to the device.

The "its result will be cast to the configured type" contradicts "In NGSIv1, no matter the input of such expression, the output always cast to String". Is the documentation wrong? In that case, it should be fixed.

If the documentation is ok, it should clarify which type is used: the one defined for the attribute in the API or the one to which the expression belongs (I mean, if the expression belong to "string operations" the type is String and if the expression belongs to "numeric operations" the type is Number).